dask_sql/physical/rex/core/call.py

```diff
@@ -960,6 +962,11 @@ def date_part(self, what, df: SeriesOrScalar):
             return df.week
         elif what in {"YEAR", "YEARS"}:
             return df.year
+        elif what == "DATE":
+            if "cudf" in str(df._meta):
+                return df.strftime("%Y-%m-%d")
+            else:
+                return dd.to_datetime(df.strftime("%Y-%m-%d"))
```

Comment:
This chunk of code allows something like `SELECT EXTRACT(DATE FROM d) + INTERVAL '2 days' FROM df` to work on the CPU, since the DataFrame needs to be converted to a datetime dtype in order to use the `+` operator. However, if it's a GPU DataFrame, then `dd.to_datetime()` causes errors. I couldn't get `cudf.to_datetime()` to work in this case, and I don't think there's a `dask_cudf.to_datetime()` function at all.

TLDR: This code allows flexibility with different operations on the CPU, but doesn't work for every GPU example. See: https://gist.github.com/sarahyurick/d5741d8a8e7f05a5ea8fcb54039bb6c5 where `df` is a CPU table and `table1` is a GPU table.

Reply:
Another note: I already have an issue https://github.com/rapidsai/cudf/issues/12062 about `dd.to_datetime` which would probably solve this error.